src/Renderer/Model/Model.cpp

```diff
@@ -0,0 +1,92 @@
+#include "Model.h"
+
+#include <cstring>
+
+namespace SnekVk
+{
+    Model::Model(VulkanDevice& device, const Vertex* vertices, u32 vertexCount)
+        : device{device}, vertexCount{vertexCount}
+    {
+        CreateVertexBuffers(vertices);
+    }
+
+    Model::~Model()
+    {
+        // Destroy the vertex buffer and free the memory
+        vkDestroyBuffer(device.Device(), vertexBuffer, nullptr);
+        vkFreeMemory(device.Device(), vertexBufferMemory, nullptr);
```

Comment:
Maybe consider in future giving the renderer a map<model, device> or even if there's only one device, use a static method to access it. Avoid forcing the model to hold the direct reference for its lifetime.